cmd/prometheus/main.go

```diff
@@ -239,6 +239,9 @@ func main() {
 	a.Flag("rules.alert.resend-delay", "Minimum amount of time to wait before resending an alert to Alertmanager.").
 		Default("1m").SetValue(&cfg.resendDelay)
 
+	a.Flag("scrape.timestamp-tolerance", "Adjust scrape timestamps by up to this amount to align them to the intended schedule. Experimental.").
+		Hidden().Default("2ms").DurationVar(&scrape.ScrapeTimestampTolerance)
```

Comment:
This seems quite a bit riskier than a simple disable option, users may put in much larger values without understanding that this flag will cause artifacts in their graphs.

We should probably also make it very clear that this flag will be removed in future. Given past history this flag will likey hang around and defacto require us to provide this as a feature indefinitely, unless we're very clear on this and have agreed criteria for removing it (i.e. how close to previous typical compression results is okay).